packages/machine-extractor/src/toMachineConfig.ts

```diff
@@ -33,6 +33,7 @@ export interface ToMachineConfigOptions {
 const parseStateNode = (
   astResult: StateNodeReturn,
   opts: ToMachineConfigOptions | undefined,
+  isRootNode: boolean,
```

Comment:
instead of adding a parameter here, I'd introduce a wrapper function that would handle `context` and then call a recursive function for handling states. This way we just introduce a more specialized handler at the top but the core handler (this one here) stays ignorant to the `isRootNode`

Reply:
Check it out again.